devicetypes/smartthings/zwave-switch-battery.src/zwave-switch-battery.groovy

```diff
@@ -0,0 +1,170 @@
+/**
+ *	Copyright 2018 SmartThings
+ *
+ *	Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *	in compliance with the License. You may obtain a copy of the License at:
+ *
+ *		http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *	Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *	on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *	for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+	definition (name: "Z-Wave Switch Battery", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "oic.d.switch", runLocally: true, minHubCoreVersion: '000.017.0012', executeCommandsLocally: true) {
+		capability "Actuator"
+		capability "Battery"
+		capability "Health Check"
+		capability "Refresh"
+		capability "Sensor"
+		capability "Switch"
+
+		fingerprint mfr:"014A", prod:"0006", model:"0005", deviceJoinName: "Smart Switch Single Rocker"
+		fingerprint mfr:"014A", prod:"0006", model:"0004", deviceJoinName: "Dual Toggle Switch"
+		fingerprint mfr:"014A", prod:"0006", model:"0003", deviceJoinName: "Smart Switch Dual Rocker"
+		fingerprint mfr:"014A", prod:"0006", model:"0002", deviceJoinName: "Single Gang Toggle Wirless Light Switch"
+	}
+
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"switch", type: "lighting", width: 6, height: 4, canChangeIcon: true){
+			tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label: '${name}', action: "switch.off", icon: "st.switches.switch.on", backgroundColor: "#00A0DC"
+				attributeState "off", label: '${name}', action: "switch.on", icon: "st.switches.switch.off", backgroundColor: "#ffffff"
+			}
+		}
+
+		standardTile("refresh", "device.switch", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+		valueTile("battery", "device.battery", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+
+		main "switch"
+		details(["switch","refresh","battery"])
+	}
+}
+
+def installed() {
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
```

Comment:
Extract sendEvent in installed and updated to one method so in future it might be changed in one place (easier to remember than two)